tests/checker/test_loginurl.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2020 Chris Mayo
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Test director.aggregator.Aggregate.visit_loginurl().
+This includes the retrieval and search of a login form and posting credentials.
+"""
+import re
+
+from .httpserver import HttpServerTest, CGIHandler
+from . import get_test_aggregate
+
+class TestLoginUrl(HttpServerTest):
+    """Test loginurl retrieval, search and posting credentials."""
+
+    def __init__(self, methodName="runTest"):
+        super().__init__(methodName=methodName)
+        self.handler = CGIHandler
```

Comment:
Now I know this is existing practice in linkchecker's test suite, but I'm mildly uncomfortable overriding `__init__` in test classes.  This could be a class attribute, if the parent class also used class attributes instead of assigning to `self.port` and `self.handler` in an overridden `__init__`.

Reply:
could we just do the **kwargs dance here to carry those around instead?

Reply:
I'm not sure what you mean, @anarcat

Reply:
suggested a change above... maybe i misunderstood your comment?

Reply:
ah, the "Conversation" tab shows these comments in a different order, and there's nothing above this one.  I see a suggested change below :)